### PR TITLE
[controller] Fix node removable check with locked instances

### PR DESCRIPTION
**Pull request by nisargthakkar**
## Fix node removable check with locked instances

Currently, there is a bug where even if an instance is non-removable, if it doesn't share any partition with some locked instance, it returns a removable result for the instance.

This commit also depracates the cluster view in the `node_removable` check and removes the `instance_view` API param.

## How was this PR tested?

Added extra checks in unit tests

## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure to explain your proposed changes and call out the behavior change.
    - Before this change, invoking `/node_removable` endpoint on controllers would check if the cluster view is healthy after removal of the specified instance. However, this is not desirable because it would block the removal of the instance even if it doesn't cause any change to the cluster's healthiness. This mode has been deprecated in favor of the `instance_view` mode that was previously available by specifying an API param. Specifying the API param is now a no-op, so, it is still a backwards compatible change

### CODE REVIEW

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/InstanceStatusDecider.java [hunk 1]
```diff
@@ -32,29 +32,20 @@ static List<Replica> getReplicasForInstance(HelixVeniceClusterResources resource
...
   private static PartitionAssignment removeLockedResources(
       PartitionAssignment partitionAssignment,
       List<String> lockedNodes,
-      String resourceName,
-      boolean isInstanceView) {
+      String resourceName) {
     PartitionAssignment assignment = partitionAssignment;
     for (String lockedNodeId: lockedNodes) {
-      assignment = getPartitionAssignmentAfterRemoving(lockedNodeId, assignment, resourceName, isInstanceView);
+      assignment = getPartitionAssignmentAfterRemoving(lockedNodeId, assignment, resourceName, false);
```
Comment (majisourav99) [inline services/venice-controller/src/main/java/com/linkedin/venice/controller/InstanceStatusDecider.java:44]: Can we add some comment here why we are passing instanceView false for locked nodes.

Comment (nisargthakkar) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/InstanceStatusDecider.java:44]: Sure. Will add one

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/InstanceStatusDecider.java [hunk 2]
```diff
@@ -241,14 +225,18 @@ private static PartitionAssignment getPartitionAssignmentAfterRemoving(
       String instanceId,
       PartitionAssignment partitionAssignment,
       String resourceName,
-      boolean isInstanceView) {
+      boolean lockedInstance) {
     PartitionAssignment partitionAssignmentAfterRemoving =
         new PartitionAssignment(resourceName, partitionAssignment.getExpectedNumberOfPartitions());
     for (Partition partition: partitionAssignment.getAllPartitions()) {
-      if (isInstanceView) {
+      if (!lockedInstance) {
```
Comment (majisourav99) [inline services/venice-controller/src/main/java/com/linkedin/venice/controller/InstanceStatusDecider.java:232]: Thanks renaming the variable, more informative

**Review by majisourav99 — APPROVED** [review]
LGTM

**Pull request merged**